internal/uvm/share.go

```diff
@@ -0,0 +1,64 @@
+package uvm
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/Microsoft/hcsshim/internal/guestrequest"
+	"github.com/Microsoft/hcsshim/internal/requesttype"
+	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
+)
+
+// Share shares in file(s) from `reqHostPath` on the host machine to `reqUVMPath` inside the UVM.
+// This function handles both LCOW and WCOW scenarios.
+func (uvm *UtilityVM) Share(ctx context.Context, reqHostPath, reqUVMPath string, readOnly bool) error {
+	if uvm.OS() == "windows" {
+		if _, err := uvm.GetVSMBUvmPath(ctx, reqHostPath, readOnly); err == ErrNotAttached {
+			// share file has not been added yet, add it now
+			options := uvm.DefaultVSMBOptions(readOnly)
+			_, err := uvm.AddVSMB(ctx, reqHostPath, options)
```

Comment:
We should take the `VSMBShare` returned and defer `Release` on it if there's an error in any of the calls below. And yes I know this is just my code from the shim share method 😆

Reply:
Done :)